types/block.go

```diff
@@ -34,13 +34,34 @@ const (
 	MaxAminoOverheadForBlock int64 = 11
 )
 
+// DataAvailabilityHeader contains the row and column roots of the erasure
+// coded version of the data in Block.Data.
+// Therefor the original Block.Data is arranged in a
+// k × k matrix, which is then "extended" to a
+// 2k × 2k matrix applying multiple times Reed-Solomon encoding.
+// For details see Section 5.2: https://arxiv.org/abs/1809.09044
+type DataAvailabilityHeader struct {
+	// RowRoot_j 	= root((M_{j,1} || M_{j,2} || ... || M_{j,2k} ))
+	RowsRoots [][]byte
+	// ColumnRoot_j = root((M_{1,j} || M_{2,j} || ... || M_{2k,j} ))
+	ColumnRoots [][]byte
+}
+
+// Hash computes the root of the row and column roots
+// TODO: feed this into Header.DataHash.
+func (dah *DataAvailabilityHeader) Hash() {
+	panic("TODO: implement")
+}
+
 // Block defines the atomic unit of a Tendermint blockchain.
 type Block struct {
-	mtx        sync.Mutex
-	Header     `json:"header"`
-	Data       `json:"data"`
-	Evidence   EvidenceData `json:"evidence"`
-	LastCommit *Commit      `json:"last_commit"`
+	mtx sync.Mutex
+
+	Header                 `json:"header"`
+	Data                   `json:"data"`
+	DataAvailabilityHeader DataAvailabilityHeader
```

Comment:
Is there a reason there's no "json:..." for this line? Is this something that's deprecated anyways?

Reply:
Hmm, these annotations are (directly) used for JSON encoding. On the long run I'd like to separate core types that have methods and hold logic  from serialization concerns. But maybe we should keep these annotations. I think in geth there are also annoations but they are used to *generate* types that purely deal with serialization (better than what we have here). For now you are right, there should als be annotations. Will add them in a followup PR (please don't resolve this comment until then).

Reply:
NVM, added them in https://github.com/LazyLedger/lazyledger-core/pull/17/commits/6373e8850fb21107fc42d703e6509fb6cf131f3d.